cylc/flow/commands.py

```diff
@@ -0,0 +1,434 @@
+        with suppress(KeyError):
+            # By default, mode from mutation is a name from the
+            # WorkflowStopMode graphene.Enum, but we need the value
+            mode = WorkflowStopMode[mode]  # type: ignore[misc]
+        try:
+            mode = StopMode(mode)
+        except ValueError:
+            raise CommandFailedError(f"Invalid stop mode: '{mode}'")
+        schd._set_stop(mode)
+        if mode is StopMode.REQUEST_KILL:
+            schd.time_next_kill = time()
+
+
+@_command('release')
+async def release(schd: 'Scheduler', tasks: Iterable[str]):
+    """Release held tasks."""
+    yield
+    yield schd.pool.release_held_tasks(tasks)
+
+
+@_command('release_hold_point')
+async def release_hold_point(schd: 'Scheduler'):
+    """Release all held tasks and unset workflow hold after cycle point,
+    if set."""
+    yield
+    LOG.info("Releasing all tasks and removing hold cycle point.")
+    schd.pool.release_hold_point()
+
+
+@_command('resume')
+async def resume(schd: 'Scheduler'):
+    """Resume paused workflow."""
+    yield
+    schd.resume_workflow()
+
+
+@_command('poll_tasks')
+async def poll_tasks(schd: 'Scheduler', tasks: Iterable[str]):
+    """Poll pollable tasks or a task or family if options are provided."""
+    yield
+    if schd.get_run_mode() == RunMode.SIMULATION:
+        yield 0
+    itasks, _, bad_items = schd.pool.filter_task_proxies(tasks)
+    schd.task_job_mgr.poll_task_jobs(schd.workflow, itasks)
+    yield len(bad_items)
+
+
+@_command('kill_tasks')
+async def kill_tasks(schd: 'Scheduler', tasks: Iterable[str]):
+    """Kill all tasks or a task/family if options are provided."""
+    yield
+    itasks, _, bad_items = schd.pool.filter_task_proxies(tasks)
+    if schd.get_run_mode() == RunMode.SIMULATION:
+        for itask in itasks:
+            if itask.state(*TASK_STATUSES_ACTIVE):
+                itask.state_reset(TASK_STATUS_FAILED)
+                schd.data_store_mgr.delta_task_state(itask)
+        yield len(bad_items)
+    else:
+        schd.task_job_mgr.kill_task_jobs(schd.workflow, itasks)
+        yield len(bad_items)
+
+
+@_command('hold')
+async def hold(schd: 'Scheduler', tasks: Iterable[str]):
+    """Hold specified tasks."""
+    yield
+    yield schd.pool.hold_tasks(tasks)
+
+
+@_command('set_hold_point')
+async def set_hold_point(schd: 'Scheduler', point: str):
+    """Hold all tasks after the specified cycle point."""
+    yield
+    cycle_point = TaskID.get_standardised_point(point)
+    if cycle_point is None:
+        raise CyclingError("Cannot set hold point to None")
+    LOG.info(
+        f"Setting hold cycle point: {cycle_point}\n"
+        "All tasks after this point will be held."
+    )
+    schd.pool.set_hold_point(cycle_point)
+
+
+@_command('pause')
+async def pause(schd: 'Scheduler'):
+    """Pause the workflow."""
+    yield
+    schd.pause_workflow()
+
+
+@_command('set_verbosity')
+async def set_verbosity(schd: 'Scheduler', level: Union[int, str]):
```

Comment:
It seems odd on a first inspection that this function (formerly an `@staticmethod` requires the scheduler arg, but it does - why?

Reply:
All of the commands must have the same interface so we can handle them in an abstract way:

```python
async def <command>(schd: 'Scheduler', *args, *kwargs):
```

Otherwise we would need to go back to having a seperate implementation of each command in the server code.

Reply:
On the whole this approach is much tidier... 😄